pkg/services/auth/jwt/key_sets.go

```diff
@@ -189,6 +221,30 @@ func (ks *keySetJWKS) Key(ctx context.Context, keyID string) ([]jose.JSONWebKey,
 	return ks.JSONWebKeySet.Key(keyID), nil
 }
 
+func (ks *keySetHTTP) getBearerToken() (string, error) {
```

Comment:
I'd suggest turning this into a function that is responsible exclusively for reading the token from the file and returning a `Bearer <file_contents>` string. It feels as though we would to store the already processed authorization header into `keySetHTTP` (unless there's a legit use case for re-reading that file periodically).

Reply:
One of the use cases for the bearer token is in Kubernetes pods.
The serviceaccount token mounted/projected in at `/var/run/secrets/kubernetes.io/serviceaccount/token` has a lifetime of either 15 minutes or a year depending on your cluster config, the latter being in the process of being phased out.
Re-reading the token is then a must 😄
But `getBearerToken` has now been refactored into a function!